akka-stream/src/main/scala/akka/stream/Materializer.scala

```diff
@@ -144,6 +145,42 @@ abstract class Materializer {
     since = "2.6.0")
   def schedulePeriodically(initialDelay: FiniteDuration, interval: FiniteDuration, task: Runnable): Cancellable
 
+  /**
+   * Shuts down this materializer and all the operators that have been materialized through this materializer. After
+   * having shut down, this materializer cannot be used again. Any attempt to materialize operators after having
+   * shut down will result in an IllegalStateException being thrown at materialization time.
+   */
+  def shutdown(): Unit
+
+  /**
+   * Indicates if the materializer has been shut down.
+   */
+  def isShutdown: Boolean
+
+  /**
+   * INTERNAL API
+   *
+   * FIXME this was documented as internal api but was public, can we make it internal?
```

Comment:
I think it should be public. No reason to hide it, especially not now when the stream is typically run with a system.

Reply:
I wonder if we should deprecate the scheduler forwarder methods as well then, since those were there when there was no public API access to the system from the materializer?

Reply:
makes sense to me